File: spring-security-login-logger/src/main/java/com/andrewthom/springsecurityloginlogger/listener/AuthenticationResult.java
package com.andrewthom.springsecurityloginlogger.listener;

import org.springframework.security.authentication.event.*;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public enum AuthenticationResult {
	SUCCESS(AuthenticationSuccessEvent.class),
	FAILURE(AbstractAuthenticationFailureEvent.class),
	BAD_CREDENTIALS(AuthenticationFailureBadCredentialsEvent.class),
	ACCOUNT_CREDENTIALS_EXPIRED(AuthenticationFailureCredentialsExpiredEvent.class),
	ACCOUNT_DISABLED(AuthenticationFailureDisabledEvent.class),
	ACCOUNT_EXPIRED(AuthenticationFailureExpiredEvent.class),
	ACCOUNT_LOCKED(AuthenticationFailureLockedEvent.class),
	PROVIDER_NOT_FOUND(AuthenticationFailureProviderNotFoundEvent.class),
	PROXY_UNTRUSTED(AuthenticationFailureProxyUntrustedEvent.class),
	AUTH_SERVICE_EXCEPTION(AuthenticationFailureServiceExceptionEvent.class);

	private static final Map<Class<? extends AbstractAuthenticationEvent>, AuthenticationResult> CLASS_TO_RESULT;
	static {
		Map<Class<? extends AbstractAuthenticationEvent>, AuthenticationResult> builder = new HashMap<>();
		for (AuthenticationResult result : values()) {
			builder.put(result.klass, result);
		}
		CLASS_TO_RESULT = Collections.unmodifiableMap(builder);
	}

	private final Class<? extends AbstractAuthenticationEvent> klass;

	AuthenticationResult(Class<? extends AbstractAuthenticationEvent> klass) {
		this.klass = klass;
	}

	public static AuthenticationResult getFailureByClass(Class<? extends AbstractAuthenticationFailureEvent> klass) {
		return CLASS_TO_RESULT.getOrDefault(klass, FAILURE);
	}
}

File: spring-security-login-logger/src/main/kotlin/com/andrewthom/springsecurityloginlogger/common/listener/AuthenticationResult.kt
package com.andrewthom.springsecurityloginlogger.common.listener

import org.springframework.security.authentication.event.*
import kotlin.reflect.KClass

enum class AuthenticationResult(private val klass: KClass<out AbstractAuthenticationEvent>) {
	SUCCESS(AuthenticationSuccessEvent::class),
	FAILURE(AbstractAuthenticationFailureEvent::class),
	BAD_CREDENTIALS(AuthenticationFailureBadCredentialsEvent::class),
	ACCOUNT_CREDENTIALS_EXPIRED(AuthenticationFailureCredentialsExpiredEvent::class),
	ACCOUNT_DISABLED(AuthenticationFailureDisabledEvent::class),
	ACCOUNT_EXPIRED(AuthenticationFailureExpiredEvent::class),
	ACCOUNT_LOCKED(AuthenticationFailureLockedEvent::class),
	PROVIDER_NOT_FOUND(AuthenticationFailureProviderNotFoundEvent::class),
	PROXY_UNTRUSTED(AuthenticationFailureProxyUntrustedEvent::class),
	AUTH_SERVICE_EXCEPTION(AuthenticationFailureServiceExceptionEvent::class);

	companion object {
		private val CLASS_TO_RESULT: Map<KClass<out AbstractAuthenticationEvent>, AuthenticationResult> = values().associateBy { it.klass }

		fun getFailureByClass(klass: KClass<out AbstractAuthenticationFailureEvent>): AuthenticationResult {
			return CLASS_TO_RESULT[klass] ?: FAILURE
		}
	}
}
